primitives/core/src/crypto.rs

```diff
@@ -239,17 +239,35 @@ pub trait Ss58Codec: Sized + AsMut<[u8]> + AsRef<[u8]> + Default {
 		let mut res = Self::default();
 		let len = res.as_mut().len();
 		let d = s.from_base58().map_err(|_| PublicError::BadBase58)?; // failure here would be invalid encoding.
-		if d.len() != len + 3 {
-			// Invalid length.
+		let prefix = if d.len() == len + 3 && d[0] < 64 {
+			1
+		} else if d.len() == len + 4 && d[0] >= 64 && d[0] < 128 {
+			2
+		} else {
```

Comment:
I can see the rough shape of what this does, but an explanatory comment would be nice :-)

Reply:
if I'm correct context is here https://github.com/paritytech/substrate/wiki/External-Address-Format-(SS58)